js/modules/k6/ws/ws.go

```diff
@@ -159,13 +159,14 @@ func (*WS) Connect(ctx context.Context, url string, args ...goja.Value) (*WSHTTP
 	}
 
 	wsd := websocket.Dialer{
-		NetDial:         netDial,
-		Proxy:           http.ProxyFromEnvironment,
-		TLSClientConfig: tlsConfig,
+		HandshakeTimeout: time.Second * 60, // TODO configurable
+		NetDial:          netDial,
+		Proxy:            http.ProxyFromEnvironment,
+		TLSClientConfig:  tlsConfig,
 	}
 
 	start := time.Now()
-	conn, httpResponse, connErr := wsd.Dial(url, header)
+	conn, httpResponse, connErr := wsd.DialContext(ctx, url, header)
```

Comment:
Hmm this might be problematic, see `netDial` above, we have `state.Dialer.DialContext(ctx...)`

Reply:
looking at the code it isn't a problem, but will change the above line as well ;)

Reply:
hmm why not just set the `NetDialContext` property of the websocket `Dialer` directly to our `DialContext`?

Reply:
that is exactly what I did :D , but the underlying code in gorilla would've done it as well ..so I am pretty sure that if anything it is the HandshakeTimeout that will help

Reply:
resolved in a13b548